### PR TITLE
Check for config file before attempting to read

### LINKED ISSUE #663: Errno::ENOENT: No such file or directory @ rb_sysopen config/webpacker.yml
The webpacker's master branch (rails v5.1.3, ruby 2.4) is giving this error when i run `rails webpacker:install` on an existing rails project. The immediate solution is to use `gem 'webpacker', '~> 2.0'` instead. 

> Errno::ENOENT: No such file or directory @ rb_sysopen - /Users/ytbryan/Desktop/X/vue-X/config/webpacker.yml
> /Users/ytbryan/Desktop/Doerlist/X/config/environment.rb:5:in `<top (required)>'
> /Users/ytbryan/Desktop/Doerlist/X/bin/rails:9:in `require'
> /Users/ytbryan/Desktop/Doerlist/X/bin/rails:9:in `<top (required)>'
> /Users/ytbryan/Desktop/Doerlist/X/bin/spring:14:in `require'
> /Users/ytbryan/Desktop/Doerlist/X/bin/spring:14:in `<top (required)>'
> ./bin/rails:3:in `load'
> ./bin/rails:3:in `<main>'
> Tasks: TOP => app:template => environment
> (See full trace by running task with --trace)
> 

I will look at this when I'm free. Feel free to report or PR if you find out what's wrong.

### PR DESCRIPTION
**Pull request by gauravtiwari**
Fixes #663 i.e. reading config file before webpacker is installed

### CODE REVIEW

**Comment by dhh** [comment]
I'd rather figure out why Webpacker is being booted at a time prior to webpacker.yml existing. It doesn't make sense to me to silently swallow a missing config file in normal use. So I'd rather just wrap the initialization on this check.

**Comment by gauravtiwari** [comment]
Oh yes, makes sense. I will investigate and update this PR 👍🙏

**Comment by gauravtiwari** [comment]
Added `defined?(Rails::Server) || defined?(Rails::Console)` check prior to bootstrap, not sure if the check is robust though. The idea is to try and bootstrap whenever server or console is run and fail (compared to config file check which would ignore bootstrap in all cases), for other cases like running a rake task don't bootstrap.

File: lib/webpacker/configuration.rb [hunk 1]
```diff
@@ -56,6 +56,16 @@ def data
 
     def load
       YAML.load(config_path.read)[env].deep_symbolize_keys
+
+    rescue Errno::ENOENT => e
+      raise "Webpacker configuration file not found #{config_path}. " \
+            "Have you installed webpacker? " \
```
Comment (ytbryan) [inline lib/webpacker/configuration.rb]: I think this line should be replaced with `Please run: rails webpacker:install`. What do you think?

**Comment by gauravtiwari** [comment]
@dhh Please review and merge this one as well if looks good :)

**Comment by dhh** [comment]
Might want to consider @ytbryan's exception message improvement as well.

File: lib/webpacker/configuration.rb (hunk 1 above)
Comment (gauravtiwari) [inline lib/webpacker/configuration.rb]: @ytbryan Oh totally missed, feel free to make a PR for this 👍

**Pull request merged**